### PR TITLE
Added custom button menu to Resource Details page

### PR DESCRIPTION
**Pull request by chalettu**
@miq-bot add_label  gaprindashvili/yes
@miq-bot add_label enhancement
BZ - https://bugzilla.redhat.com/show_bug.cgi?id=1513113

### CODE REVIEW

**Comment by chalettu** [comment]
Before SS 
before_ss

**Comment by chalettu** [comment]
After SS
new_menu

**Comment by Loicavenel** [comment]
@AllenBW I am confused here. the design if I am not mistaking is to have the kebab menu when there are too many Custom button on the top bar not all the time...

**Comment by Loicavenel** [comment]
@serenamarie125 need your comment here..

**Comment by chalettu** [comment]
@Loicavenel , no need to ask @AllenBW , I was the one working on this issue.  This kebab menu will not show all the time, we put a limit of 3 custom buttons or button groups to show on the screen before we show the kebab menu.

**Comment by Loicavenel** [comment]
@chalettu my apologise..

**Comment by chalettu** [comment]
No worries @Loicavenel .  Screenshots and small videos can sometimes make it hard to convey other logic behind what is intended.

**Comment by chriskacerguis** [comment]
@chalettu could you please address the CC issues?

@AllenBW I think what @chalettu is showing is that it is limiting the buttons now (it's not all the time)...at least that is not what the desired behavior is.  @chalettu could you clarify?

**Comment by chalettu** [comment]
@serenamarie125 , I agree that button spacing is an issue.  Chris K had created another issue to address it for clarity sake and being consistent.  
SS is just an abbreviation for screen shot.  So the screenshots are for demo purposes.  The code as implemented will only default to kebab menu after 3 buttons/button groups are defined.  We can change this count if you like.  
The tooltips are aren't necessary and we can remove them if you feel they don't fit for that drop down.  Please let me know and we can definitely have them removed.

**Comment by serenamarie125** [comment]
LOL - i'm being too literal sorry @chalettu now i get the "SS"

tooltips are fine, @Loicavenel just showed me that you can set hover text, so that makes sense.

I do see an issue that the drop down menu has a caret at the top ... it should be straight, similar to the user menu shown below
image

**Comment by chalettu** [comment]
@serenamarie125 , Here is a SS with the caret removed.  Please let me know if we think any more changes might be required.  
testing

**Comment by serenamarie125** [comment]
@chalettu this is really more complex :\

Kebab uses caret (http://www.patternfly.org/pattern-library/widgets/#kebabs)
- note that the kebab has a selected state and the caret should be centered - this screenshot is from the code example from the link above
screen shot 2017-11-20 at 8 13 23 pm


The regular menus ( snapshots & power operations) shouldn't

**Comment by AllenBW** [comment]
@chalettu Any update on centering of the kebab dropdown ?

**Comment by chalettu** [comment]
@AllenBW , haven't come back to this one.  I will try and take a look at centering the kebab.

**Comment by AllenBW** [comment]
@chalettu I suspect this might be a less is more kinda situation... rip out classes see what changes... 🤔

**Comment by chalettu** [comment]
Updated SS where I added back the caret and centered it. 
updated_caret

**Comment by chalettu** [comment]
@serenamarie125 , can you take a look at this latest screenshot and see if it matches your expectations of this caret menu?

**Comment by serenamarie125** [comment]
@miq-bot add_labels ux/approved

**Comment by serenamarie125** [comment]
@miq-bot remove_labels ux/review

**Comment by serenamarie125** [comment]
@miq-bot add_label ux/approved

**Comment by serenamarie125** [comment]
@miq-bot remove_label ux/review

File: client/app/services/custom-button-menu/custom-button-menu.component.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    function postSuccess (response) {
+      if (response.success === false) {
+        EventNotifications.error(response.message)
+      } else {
+        EventNotifications.success(response.message)
+      }
+    }
+
+    function postFailure () {
```
Comment (AllenBW) [inline client/app/services/custom-button-menu/custom-button-menu.component.js]: There's gonna be real valuable server feedback on the error here, [something similar to this](https://github.com/manageiq/manageiq-ui-service/blob/01c16e684a29a8aba449249b8028c79f223eead3/client/app/services/process-snapshots-modal/process-snapshots-modal.component.js) should be used here

**Comment by chalettu** [comment]
@AllenBW , I updated based on feedback. Let me know if you approve.

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit 0b8e5730075e7feead9d824be159191f2124679a
Author: Allen Wight <allen.b.wight@gmail.com>
Date:   Fri Dec 8 07:10:47 2017 -0500

    Merge pull request #1254 from chalettu/custom-button-grouping
    
    Added custom button menu to Resource Details page
    (cherry picked from commit 672d1281341c8e7f818d1779682679e9727abfa4)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1524718
```

**Pull request merged**